### DISCUSSION TITLE
Glob patterns not working for me

### DISCUSSION THREAD

**Discussion by jchiatt** [community/community#26145] [Actions]
<p>Howdy!</p>
<p>I am working on a project in a monorepo. Each package is an individual static site generated by <a href="https://www.11ty.dev/" rel="nofollow noopener">Eleventy</a>, and we are deploying to <a href="https://workers.cloudflare.com/sites" rel="nofollow noopener">CloudFlare Workers Sites</a>.</p>
<p>All of the sites follow a very similar structure and have the same config, so I’m using <a href="https://www.hygen.io/" rel="nofollow noopener">hygen</a> to be able to generate all the boilerplate for a new site easily. I’m also hoping to use this to generate a Workflow for each site’s staging and prod environments.</p>
<p>On each Pull Request, I want to build and deploy staging instances of whatever package(s) (sites) I was working on for that PR. Initially, I thought I could do this by pattern matching the branch name but I had issues getting the glob pattern to match. The idea being I could prepend any branch related to <code>site-a</code> with <code>site-a</code> like this: <code>feat/site-a/add-the-thing</code>.</p>
<p>Example:</p>
<pre><code>on:
  pull_request:
    branches:
      - ' **site-a**'
</code></pre>
<p>^ does not work. </p>
<p>But then I read about <code>paths</code> and thought that would fit my use case much better, especially since it detects files that have been modified inside the directories that match your glob patterns.</p>
<p>So I tried:</p>
<pre><code>on:
  pull_request:
    paths:
    - ' **site-a**'
</code></pre>
<p>But to no avail. When I open a PR, it triggers, but it also seems to trigger even when_ no files were modified inside a path containing <code>site-a</code>._</p>
<p>Here’s the full config for site-a’s staging workflow right now:</p>
<pre><code>name: Deploy Site A Homepage to Workers Staging

on:
  pull_request:
    paths:
      - ' **site-a**'

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Setup Node
        uses: actions/setup-node@v1
        with:
          node-version: "12.x"

      - name: Install Dependencies
        run: yarn install
        working-directory: ./packages/sites/regions/us-west/site-a/home

      - name: Build
        run: yarn build
        working-directory: ./packages/sites/regions/us-west/site-a/home

      - name: Deploy
        run: yarn run wrangler publish --env staging
        working-directory: ./packages/sites/regions/us-west/site-a/home
</code></pre>

**Comment by jbeveland27** [comment]
<p>Did you try this by chance? </p>
<pre><code>on:
  pull_request:
    branches:
    - ' **/site-a/**'
</code></pre>
<p>That pattern might match what you’re looking to accomplish if I’m not mistaken.</p>
<p>If that fails, maybe you could try using the if conditional at the job level (which is admittedly cumbersome if you have several jobs).</p>
<p><a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif</a></p>

**Comment by jbeveland27** [comment]
<p>Did you try this by chance? </p>
<pre><code>on:
  pull_request:
    branches:
    - ' **/site-a/**'
</code></pre>
<p>That pattern might match what you’re looking to accomplish if I’m not mistaken.</p>
<p>If that fails, maybe you could try using the if conditional at the job level (which is admittedly cumbersome if you have several jobs).</p>
<p><a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif</a></p>

**Comment by jchiatt** [comment]
<p>Update: I found <a href="https://github.community/t/on-push-path-doesnt-work/17378/6">this thread</a> and am I understanding correctly that my <code>paths</code> pattern will indeed be matched against the <em>entire branch history</em> rather than just asking “what’s changed since the last push” ?</p>
<p>If so, that’s really unfortunate for my situation. </p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/jchiatt">@jchiatt</a> ,</p>
<p>For ‘pull_request’ event, the default types are: opened, synchronize, and reopened. </p>
<p>But to no avail. When I open a PR, it triggers, but it also seems to trigger even when_ no files were modified inside a path containing <code>site-a</code>._</p>
<p>_&gt;&gt; _If you triggered a workflow from PR but not close PR, edit a ‘not-site-a’ file,  the workflow will still be triggered since event type ‘synchronize’ is met, and ‘paths’ fiter check the whole PR  files including the previous ‘site-a’ file.</p>
<p>Hence, you can specify the type together with ‘paths’ to avoid this situation, code sample as below:</p>
<pre><code>on:
  pull_request:
    types: opened
    paths:
      - ' **site-a**'
</code></pre>
<p>Hope it helps!</p>

**Comment by jchiatt** [comment]
<p>This makes sense to me now! Thanks for the help.</p>
